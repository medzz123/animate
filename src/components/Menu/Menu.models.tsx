export interface MenuProps {
  items?: MenuItem[];
}

interface MenuItem {
  /**
   * Label to be displayed inside the menu
   */
  label: string;
  /**
   * onClick handler triggered when label is clicked
   */
  onClick: () => void;
}
