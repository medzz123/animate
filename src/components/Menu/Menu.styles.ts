import { Content, Item, Trigger } from '@radix-ui/react-dropdown-menu';
import styled from 'styled-components';

import { tokens } from '../../theme/tokens';

export const StyledTrigger = styled(Trigger)`
  color: ${(p) => p.theme.headline};
  padding: ${tokens.sizes[4]};
  border-radius: ${tokens.sizes[4]};
  box-shadow: inset 0 0 0 ${tokens.sizes[2]} ${(p) => p.theme.headline};
`;

export const StyledContent = styled(Content)`
  min-width: ${tokens.sizes['3xs']};
  background-color: ${(p) => p.theme.background};
  border-radius: ${tokens.sizes[6]};
  padding: ${tokens.sizes[4]};
  margin-top: ${tokens.sizes[12]};
  box-shadow: 0 0 ${tokens.sizes[12]} ${tokens.sizes[4]}
    ${(p) => p.theme.headline}${tokens.alpha[8]};
`;

export const StyledItem = styled(Item)`
  font-size: ${tokens.sizes[13]};
  padding: ${tokens.sizes[5]} ${tokens.sizes[10]};
  border-radius: ${tokens.sizes[3]};
  cursor: default;
  color: ${(p) => p.theme.headline};

  &:focus {
    outline: none;
    background-color: ${(p) => p.theme.button};
    color: ${(p) => p.theme.background};
  }
`;
