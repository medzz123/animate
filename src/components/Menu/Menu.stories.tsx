import { Meta, Story } from '@storybook/react/types-6-0';

import Menu from './Menu';

export default {
  title: 'Components/Menu',
  component: Menu,
} as Meta;

const Template: Story = (storyArguments) => (
  <Menu
    items={[
      { label: 'cut', onClick: () => console.log('cut') },
      { label: 'paste', onClick: () => console.log('paste') },
      { label: 'reload', onClick: () => console.log('reload') },
    ]}
    {...storyArguments}
  />
);

export const MenuStory = Template.bind({});
