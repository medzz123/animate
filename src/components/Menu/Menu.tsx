import { Root } from '@radix-ui/react-dropdown-menu';
import { FunctionComponent } from 'react';
import { GiHamburgerMenu } from 'react-icons/gi';

import { MenuProps } from './Menu.models';
import { StyledContent, StyledItem, StyledTrigger } from './Menu.styles';

const Menu: FunctionComponent<MenuProps> = (props) => {
  const { items } = props;

  return (
    <Root data-testid="menu">
      <StyledTrigger>
        <GiHamburgerMenu size={24} />
      </StyledTrigger>
      <StyledContent>
        {items?.map((item) => (
          <StyledItem key={item.label} onSelect={item.onClick}>
            {item.label}
          </StyledItem>
        ))}
      </StyledContent>
    </Root>
  );
};

export default Menu;
