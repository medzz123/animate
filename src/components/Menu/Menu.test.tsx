import React from 'react';

import { render } from '../../../__tests__/testUtils';
import Menu from './Menu';

test('Menu Test', () => {
  const { container } = render(
    <Menu
      items={[
        {
          label: 'Load',
          onClick: jest.fn,
        },
        {
          label: 'Target',
          onClick: jest.fn,
        },
        {
          label: 'Export',
          onClick: jest.fn,
        },
        {
          label: 'Help',
          onClick: jest.fn,
        },
      ]}
    />
  );

  expect(container).toMatchSnapshot();
});
