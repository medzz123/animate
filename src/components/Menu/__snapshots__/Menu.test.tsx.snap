// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Test 1`] = `
<div>
  <button
    aria-haspopup="menu"
    class="sc-bdvvaa eDPJOX"
    data-state="closed"
    type="button"
  >
    <svg
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
      />
    </svg>
  </button>
</div>
`;
